tests/lib/rules/no-fallthrough.js

```diff
@@ -297,6 +297,18 @@ ruleTester.run("no-fallthrough", rule, {
                     column: 34
                 }
             ]
+        },
+        {
+            code: "switch (foo) { case 0: a(); \n// eslint-enable no-fallthrough\n case 1: }",
```

Comment:
Can we add more test cases with other disable directives as well - `eslint-disable`, `eslint-disable-next-line`. Would be great to include the same case mentioned in the issue 

```js
switch (test) {
  case a:
    hello();

  // eslint-disable-next-line no-fallthrough
  case b:
}
```

Reply:
If I had used a disable directive (instead of enable), the test would have passed without my changes.

I can add them, but I'm not sure I see as much because a disable directive and a comment pattern matching a disable directive both would cause no warning to be emitted.

Reply:
@snitin315 is there a way to check for unused directives in these tests and report them as errors?

Otherwise, I don't really see how adding the OP case would be of too much help given my earlier comment.

Reply:
@gfyoung The above case will be a part of valid test cases (in this file) in case of no errors. Also, we should add a test case to assert that we don't get unused disable directive warning [here](https://github.com/eslint/eslint/blob/3e34418b31664decfb2337de798feafbf985b66c/tests/lib/linter/linter.js#L3994) like in OP, right?

Reply:
> Also, we should add a test case to assert that we don't get unused disable directive warning [here](https://github.com/eslint/eslint/blob/3e34418b31664decfb2337de798feafbf985b66c/tests/lib/linter/linter.js#L3994) like in OP, right?

That was exactly my question, and your answer is what I was looking for.  Thanks @snitin315.  Added.